cognite_toolkit/_cdf_tk/load/_resource_loaders.py

```diff
@@ -1899,3 +1909,114 @@ def count(self, ids: SequenceNotStr[NodeId]) -> int:
+        return WorkflowOrchestrationAcl(
+            [WorkflowOrchestrationAcl.Action.Read, WorkflowOrchestrationAcl.Action.Write],
+            WorkflowOrchestrationAcl.Scope.All(),
+        )
+
+    @classmethod
+    def get_id(cls, item: Workflow | WorkflowUpsert) -> str:
+        if item.external_id is None:
+            raise ValueError("Workflow must have external_id set.")
+        return item.external_id
+
+    def load_resource(self, filepath: Path, ToolGlobals: CDFToolConfig, skip_validation: bool) -> WorkflowUpsertList:
+        resource = load_yaml_inject_variables(filepath, {})
+
+        workflows = [resource] if isinstance(resource, dict) else resource
+        return WorkflowUpsertList.load(workflows)
+
+    def retrieve(self, ids: SequenceNotStr[str]) -> WorkflowList:
+        workflows = []
+        for ext_id in ids:
+            workflow = self.client.workflows.retrieve(external_id=ext_id)
+            if workflow:
+                workflows.append(workflow)
+        return WorkflowList(workflows)
+
+    def _upsert(self, items: WorkflowUpsert | WorkflowUpsertList) -> WorkflowList:
+        upserts = [items] if isinstance(items, WorkflowUpsert) else items
+        return WorkflowList([self.client.workflows.upsert(upsert) for upsert in upserts])
+
+    def create(self, items: WorkflowUpsert | WorkflowUpsertList) -> WorkflowList:
+        return self._upsert(items)
+
+    def update(self, items: WorkflowUpsertList) -> WorkflowList:
+        return self._upsert(items)
+
+
+@final
+class WorkflowVersionLoader(
+    ResourceLoader[
+        WorkflowVersionId, WorkflowVersionUpsert, WorkflowVersion, WorkflowVersionUpsertList, WorkflowVersionList
+    ]
+):
+    api_name = "workflows.versions"
+    folder_name = "workflows"
+    filename_pattern = r"^.*\.?(WorkflowVersion)$"
+    resource_cls = WorkflowVersion
+    resource_write_cls = WorkflowVersionUpsert
+    list_cls = WorkflowVersionList
+    list_write_cls = WorkflowVersionUpsertList
+    dependencies = frozenset({WorkflowLoader})
+
+    @classmethod
+    def get_required_capability(cls, items: WorkflowVersionUpsertList) -> Capability:
+        return WorkflowOrchestrationAcl(
+            [WorkflowOrchestrationAcl.Action.Read, WorkflowOrchestrationAcl.Action.Write],
+            WorkflowOrchestrationAcl.Scope.All(),
+        )
+
+    @classmethod
+    def get_id(cls, item: WorkflowVersion | WorkflowVersionUpsert) -> WorkflowVersionId:
+        return item.as_id()
+
+    def load_resource(
+        self, filepath: Path, ToolGlobals: CDFToolConfig, skip_validation: bool
+    ) -> WorkflowVersionUpsertList:
+        resource = load_yaml_inject_variables(filepath, {})
+
+        workflowversions = [resource] if isinstance(resource, dict) else resource
+        return WorkflowVersionUpsertList.load(workflowversions)
+
+    def retrieve(self, ids: SequenceNotStr[WorkflowVersionId]) -> WorkflowVersionList:
+        return self.client.workflows.versions.list([id for id in ids])
```

Comment:
Unpacking this as a list does nothing. 
```suggestion
        return self.client.workflows.versions.list(ids)
```

Reply:
And if `mypy` gives you trouble you can always create a list
```suggestion
        return self.client.workflows.versions.list(list(ids))
```